jupyter_book/__init__.py

```diff
@@ -39,6 +45,15 @@ def setup(app):
     app.connect("config-inited", add_static_files)
 
     # Directives
+    app.add_node(
+        TableOfContentsNode,
+        override=True,
+        html=(skip, None),
```

Comment:
does this need `singlehtml=(skip, None)`?

I tried adding this and now get the error

```
NotImplementedError: Unknown node: fontawesome
```

the `singlehtml` builder must use a different `html` writer that doesn't include a lot of `nodes` supported by some of the extensions perhaps?

Reply:
I can't find `sphinx_fontawesome` -- where is this imported?

It is clearly used on the `intro.md` page

Reply:
we must not be using `sphinx_fontawesome` it seems like a pretty old package. Is this enabled via `javascript`?

Reply:
ah ha -- `sphinx_panels` (https://sphinx-panels.readthedocs.io/en/latest/#inline-icons)

Reply:
ah ha -- `sphinx_panels` (https://sphinx-panels.readthedocs.io/en/latest/#inline-icons)

Reply:
I also tried adding the nodes in `sphinx_panels` to `singlehtml` in https://github.com/executablebooks/sphinx-panels/pull/61 but ended with a node I am unsure of where it resides:

```
NotImplementedError: Unknown node: _TabInput
```


It looks like the `singlehtml` builder has a long chain of unsupported `nodes`.

Reply:
@chrisjsewell any ideas where `_TabInput` comes from? I'll keep digging. Is the only way around these issues to get `singlehtml` builder supported (in similar fashion to `html`)?

Reply:
I can't find it in any of the `_build/.doctrees`

Reply:
you spent all that time looking at doctrees, when you could have just asked github 😜 https://github.com/search?q=_TabInput&type=code (its sphinx-inline-tabs)

Reply:
oh cool -- thanks -- I tried searching `github` but restricted it to `executablebooks` -- doh!

The searching through `doctrees` was:

```
grep -r "TabInput" _build/.doctrees/
```

so didn't consume too much time :-).

Reply:
Surprised that works, given that doctrees are (pickled) binary files

Reply:
yeah I was too but then I realised that a pickle is really `Pickling is a way to convert a python object (list, dict, etc.) into a character stream` so it picks up keywords. I have found it to be very useful :-)